h/api/storage.py

```diff
@@ -49,6 +49,15 @@ def fetch_annotation(session, id_):
         return None
 
 
+def fetch_ordered_annotations(session, ids):
+    if not ids:
+        return []
+
+    anns = session.query(models.Annotation).filter(models.Annotation.id.in_(ids)).all()
+    anns = sorted(anns, key=lambda a: ids.index(a.id))
```

Comment:
This is going to be crazy slow if the list is too large, because `.index` is an O(N) operation.

Probably way better to built a lookup table first as:

```python
ordering = {x: i for i, x in enumerate(ids)}
```

and then use

```python
anns = sorted(anns, key=ordering.get)
```